nbgrader/tests/ui-tests/course_list.spec.ts

```diff
@@ -0,0 +1,252 @@
+import { test as jupyterLabTest, galata, IJupyterLabPageFixture, expect } from "@jupyterlab/galata";
+
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+
+import { test as notebookTest } from './utils/notebook_fixtures';
+import { createEnv } from "./utils/test_utils";
+
+const testDir = process.env.NBGRADER_TEST_DIR || '';
+if (!testDir){
+  throw new Error('Test directory not provided');
+}
+if (!fs.existsSync(testDir)){
+  throw new Error(`Test directory ${testDir} doesn't exists`);
+}
+
+const isWindows = os.platform().startsWith('win');
+
+const tempPath = 'nbgrader-course-list-test';
+
+let test = jupyterLabTest;
+let mainPanelId = '#jp-main-dock-panel';
+let menuPanelId = '#jp-menu-panel';
+let mainPanelTabCount = 1;
+
+const baseTestUse = {
+  tmpPath: tempPath,
+  mockSettings: {
+    '@jupyterlab/apputils-extension:notification': {
+      fetchNews: 'false'
```

Comment:
Quick heads-up there were some updates on this config in core JupyterLab in https://github.com/jupyterlab/jupyterlab/pull/14609.

We can check whether the default `mockSettings` from JupyterLab is already automatically applied, if we should add `checkForUpdates: false` as well.